rosenpass/src/cli.rs

```diff
@@ -119,6 +126,45 @@ impl Cli {
                 config::Rosenpass::example_config().store(config_file)?;
             }
 
+            /// Deprecated - use gen-keys instead
+            Keygen {
+                args,
+            } => {
+                eprintln!("WARNING: The 'keygen' command is deprecated. Please use the 'gen-keys' command instead.");
+
+                let mut public_key: Option<PathBuf> = None;
+                let mut private_key: Option<PathBuf> = None;
+
+                // Manual arg parsing, since clap wants to prefix flags with "--"
+                let mut args = args.into_iter();
+                loop {
+                    match (args.next(),args.next()) {
+                        (Some(flag),Some(opt)) if flag == "private-key" => {
+                            private_key  = Some(opt.into());
+                        }
+                        (Some(flag),Some(opt)) if flag == "public-key" => {
+                            public_key  = Some(opt.into());
+                        }
+                        (Some(flag),_) => {
+                                bail!("Unknown option `{}`",flag );
+                        },
+                        (_,_) => break,
+                    };
+                }
+
+                if private_key.is_none(){
+                    bail!("private-key is required");
+                }
+                if public_key.is_none() {
+                    bail!("public-key is required");
+                }
+
+                let (mut ssk, mut spk) = (SSk::random(), SPk::random());
+                StaticKem::keygen(ssk.secret_mut(), spk.secret_mut())?;
```

Comment:
Looks great! Could you please avoid the code duplication and factor the key generation code itself out into a function and use that in Keygen as well as GenKeys? Other than that we can merge!

Reply:
Thanks. I've factored out the key generation to a function.